src/actions/LoginActions.tsx

```diff
@@ -272,7 +272,17 @@ export function logoutRequest(navigation: NavigationBase, nextLoginId?: string):
     Airship.clear()
     dispatch({ type: 'LOGOUT', data: { nextLoginId } })
     if (typeof account.logout === 'function') await account.logout()
-    navigation.navigate('login', { experimentConfig: await getExperimentConfig() })
+
+    const experimentConfig = await getExperimentConfig()
+    navigation.navigate('login', { experimentConfig })
+
+    // The floating nav-bar hack in the SceneWrapper will call
+    // `preventDefault` on the logout navigation if the keyboard is up.
+    // The hack is supposed to retry the navigation, but that's not working
+    // right for some reason. Instead, do the navigation again ourselves:
+    if (Keyboard.isVisible()) {
+      setTimeout(() => navigation.navigate('login', { experimentConfig }), 1000)
+    }
```

Comment:
Try putting a setTimeout in the SceneWrapper where the hack is located to keep things tidy. If that works, then that's preferable. If not, then we can make do with this if it works.